internal/patroni/config.go

```diff
@@ -552,7 +552,10 @@ func instanceYAML(
 
 	if !ClusterBootstrapped(cluster) {
 		// if restore status exists, then a restore occurred an the "existing" method is used
-		if cluster.Status.PGBackRest != nil && cluster.Status.PGBackRest.Restore != nil {
+		if (cluster.Status.PGBackRest != nil && cluster.Status.PGBackRest.Restore != nil) ||
+			(cluster.Spec.DataSource != nil && cluster.Spec.DataSource.ExistingVolumes != nil &&
+				cluster.Spec.DataSource.ExistingVolumes.ExistingPGDataVolume != nil &&
+				cluster.Spec.DataSource.ExistingVolumes.ExistingPGDataVolume.Directory != "") {
```

Comment:
Setting the directory value is what controls whether or not the move Job is run. For example, going from PGO v4 to v5, we don't want to run the pg_wal move job because the directory is already in the necessary path for v5. Other circumstances I could think of would be similar, where the `Directory` value is not set because a move does not need to be performed. Perhaps this could be named differently to better describe the behavior?
The comment on the `Directory` value, 
```
// The existing directory. When not set, a move Job is not created for the
// associated volume.
```
is my attempt to call that functionality out.